docs/decisions/0008-application_state_differenciation.rst

```diff
@@ -0,0 +1,176 @@
+#. Invalidate queries when user interaction results in an API data change to avoid stale data
+    * If the query is not invalidated, the re-fetching of the data is entirely dependent on the length of time set for the ``staleTime``
+
+Consequences
+************
+
+Based on the call you're making, it may be challenging to coalesce all the required attributes to make the appropriate API call.
+For example, in crafting a solution for the example stated, the components themselves did not have all the required metadata to make the API call.
+Taking advantage of ``mapStateToProps`` to retrieve base metadata around the customer configuration of the learner portal and enterprise customer information (slug, uuid) should simplify the implementation.
+
+.. code-block:: javascript
+
+    import { connect } from 'react-redux';
+    import PropTypes from 'prop-types';
+
+    const TestComponent = ({
+      enterpriseUUID,
+      enterpriseFeatures,
+      enablePortalLearnerCreditManagementScreen,
+    }) => {
+      // useEnterpriseBudgets leverages the useQuery method
+      const { data: budgetsOverview } = useEnterpriseBudgets({
+        enterpriseId: enterpriseUUID,
+        enablePortalLearnerCreditManagementScreen,
+        isTopDownAssignmentEnabled: enterpriseFeatures.topDownAssignmentRealTimeLcm,
+      });
+
+      const {
+        budgets = [],
+      } = budgetsOverview || {};
+
+      if(budgets.length === 0) {
+        return (<div>No Budgets</div>);
+      }
+
+      return (
+        <div>
+          {
+            budgets.map(budget => (
+                <p>{budget.title}</p
+              )
+            )
+          }
+        </div>
+      )
+    };
+
+    const mapStateToProps = state => ({
+      enterpriseUUID: state.portalConfiguration.enterpriseId,
+      enterpriseFeatures: state.portalConfiguration.enterpriseFeatures,
+      enablePortalLearnerCreditManagementScreen: state.portalConfiguration.enablePortalLearnerCreditManagementScreen,
+    });
+
+    TestComponent.propTypes = {
+      enterpriseUUID:PropTypes.string.isRequired,
+      enterpriseFeatures: PropTypes.shape({
+        topDownAssignmentRealTimeLcm: PropTypes.bool,
+      }).isRequired,
+      enablePortalLearnerCreditManagementScreen: PropTypes.bool.isRequired,
+    };
+
+    export default connect(mapStateToProps)(TestComponent);
+
+When writing tests for a hook that leverages useQuery, certain boilerplate code is required. Below is an example of how writing tests with a useQuery directly in the component in the form of a hook,
```

Comment:
I think it's useful to callout test implications, but not sure if a complete code example is needed in the ADR? Would simply calling out the need to wrap hook/component with `QueryClientProvider` when using react-query, along with the mention of `queryClient` test  utility function. Perhaps we link out to code snippet as the full example, but keep the textual info around it brief? 🤔

Reply:
I parred down the example of the bare minimum required to construct the wrapper as opposed to providing the example component and test file.